src/textUtils.c

```diff
@@ -59,39 +59,70 @@ size_t str_formatAdaAmount(uint64_t amount, char* out, size_t outSize)
 	return rawSize + suffixLength;
 }
 
-size_t str_formatUint64(uint64_t number, char* out, size_t outSize)
+static size_t stringifyUint64ToBufferReverse(uint64_t number, char* ptr, char* end)
 {
-	ASSERT(outSize < BUFFER_SIZE_PARANOIA);
-
-	char scratchBuffer[30];
-	char* ptr = BEGIN(scratchBuffer);
-	char* end = END(scratchBuffer);
-
 	// We print in reverse
 	// We want at least one iteration
-	int place = 0;
+	size_t printedChars = 0;
 	do {
 		WRITE_CHAR(ptr, end, '0' + (number % 10));
 		number /= 10;
-		place++;
+		++printedChars;
 	} while (number > 0);
+	return printedChars;
+}
 
-	// Size without terminating character
-	STATIC_ASSERT(sizeof(ptr - scratchBuffer) == sizeof(size_t), "bad size_t size");
-	size_t rawSize = (size_t) (ptr - scratchBuffer);
-
-	if (rawSize + 1 > outSize) {
+static void printReversedStringToBuffer(const char* reverted, size_t revertedSize, char* out, size_t outSize)
+{
+	if (revertedSize + 1 > outSize) {
 		THROW(ERR_DATA_TOO_LARGE);
 	}
-
-	// Copy reversed & append terminator
-	for (size_t i = 0; i < rawSize; i++) {
-		out[i] = scratchBuffer[rawSize - 1 - i];
+	for (size_t i = 0; i < revertedSize; i++) {
+		out[i] = reverted[revertedSize - 1 - i];
 	}
-	out[rawSize] = 0;
+	out[revertedSize] = 0;
+
+	ASSERT(strlen(out) == revertedSize);
+}
 
-	ASSERT(strlen(out) == rawSize);
+size_t str_formatUint64(uint64_t number, char* out, size_t outSize)
+{
+	ASSERT(outSize < BUFFER_SIZE_PARANOIA);
+
+	char scratchBuffer[30];
+	char* ptr = BEGIN(scratchBuffer);
+	char* end = END(scratchBuffer);
+
+	// Size without terminating character
+	const size_t rawSize = stringifyUint64ToBufferReverse(number, ptr, end);
+	printReversedStringToBuffer(scratchBuffer, rawSize, out, outSize);
+	return rawSize;
+}
 
+size_t str_formatInt64(int64_t number, char* out, size_t outSize)
+{
+	ASSERT(outSize < BUFFER_SIZE_PARANOIA);
+
+	char scratchBuffer[30];
+	char* ptr = BEGIN(scratchBuffer);
+	char* end = END(scratchBuffer);
+
+	uint64_t signlessNumber;
+	if (number < 0) {
+		if (INT64_MIN == number) {
+			signlessNumber = ((uint64_t)INT64_MAX) + 1;
+		} else {
+			signlessNumber = (uint64_t) llabs(number);
+		}
+	} else {
+		signlessNumber = (uint64_t) number;
+	}
+	// Size without terminating character (and potential sign)
+	size_t rawSize = stringifyUint64ToBufferReverse(signlessNumber, ptr, end - 1);
```

Comment:
I think we mostly use "length" and not "size" when dealing with text strings. In particular, size should be length + 1.